packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -316,13 +318,27 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
             color: game.backgroundColor(),
             child: LayoutBuilder(
               builder: (_, BoxConstraints constraints) {
-                game.onGameResize(constraints.biggest.toVector2());
+                final size = constraints.biggest.toVector2();
+                if (size.isZero()) {
+                  return widget.loadingBuilder?.call(context) ?? Container();
```

Comment:
I have little knowledge of how the `loadingBuilder` works, will this return the currently running loading widget, or will it create a new one here?

Reply:
`loadingBuilder` is a function that creates and returns a Flutter widget on each invocation. This is perfectly normal from Flutter's standpoint: everything starting from line 270 above will create a new Flutter widget every time the build method is called. If you need some persistency in widget's data, you'd declare it as a `StatefulWidget` and keep its state as a separate object whose lifetime is managed more carefully.

This is of course very different from what we do in FCS, where components are added and removed from the game explicitly.